source/simulator/core.cc

```diff
@@ -989,6 +989,8 @@ namespace aspect
   template <int dim>
   void Simulator<dim>::setup_dofs ()
   {
+    signals.edit_parameters(*this, parameters);
+
```

Comment:
I don't care one way or the other, but I think if you call it `edit_parameters` then either once or each time step seems reasonable. If you have good reasons to do it at the start of `setup_dofs()`, then call it `pre_setup_dofs()` :-)